src/CSVChannel.cpp

```diff
@@ -168,6 +184,25 @@ void CSVWriter::writeUInt16(std::string name, uint16_t value) { write(value); }
 void CSVWriter::writeUInt24(std::string name, uint32_t value) { write(value); }
 void CSVWriter::writeUInt32(std::string name, uint32_t value) { write(value); }
 
+void CSVWriter::writeHex8(std::string name, uint8_t value)
+{
+    std::stringstream ss;
+    ss << std::hex << std::uppercase << std::setfill('0') << std::setw(2) << static_cast<uint32_t>(value);
+    write(ss.str());
+}
+void CSVWriter::writeHex16(std::string name, uint16_t value)
+{
+    std::stringstream ss;
+    ss << std::hex << std::uppercase << std::setfill('0') << std::setw(4) << static_cast<uint32_t>(value);
+    write(ss.str());
+}
+void CSVWriter::writeHex32(std::string name, uint32_t value)
+{
+    std::stringstream ss;
+    ss << std::hex << std::uppercase << std::setfill('0') << std::setw(8) << static_cast<uint32_t>(value);
+    write(ss.str());
+}
```

Comment:
```suggestion
void CSVWriter::writeHex8(std::string name, uint8_t value)
{
    std::stringstream ss;
    ss << std::hex << std::uppercase << std::setfill('0') << std::setw(2) << static_cast<uint32_t>(value);
    write(std::quoted(ss.str(), '\"', '\"'));
}
void CSVWriter::writeHex16(std::string name, uint16_t value)
{
    std::stringstream ss;
    ss << std::hex << std::uppercase << std::setfill('0') << std::setw(4) << static_cast<uint32_t>(value);
    write(std::quoted(ss.str(), '\"', '\"'));
}
void CSVWriter::writeHex32(std::string name, uint32_t value)
{
    std::stringstream ss;
    ss << std::hex << std::uppercase << std::setfill('0') << std::setw(8) << static_cast<uint32_t>(value);
    write(std::quoted(ss.str(), '\"', '\"'));
}
```

By forcing it to be interpreted as text inside the CSV you prevent software like LibreOffice from accidentally detecting it as decimal number of exponential number.

Just ran into that problem somewhere else...

Reply:
Appreciated very much for sharing your experience